app/src/main/kotlin/io/imhungry/ui/MyMapActivity.kt

```diff
@@ -0,0 +1,303 @@
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import io.imhungry.GoogleAPI.IGoogleAPIService
+import io.imhungry.GoogleAPI.initiateGoogleAPIService
+import io.imhungry.model.RootObject
+import kotlinx.android.synthetic.main.activity_map.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.lang.StringBuilder
+import io.imhungry.R
+
+
+class MyMapsActivity : AppCompatActivity(), OnMapReadyCallback {
+
+
+    //For Current Location
+    private lateinit var mMap: GoogleMap
+    private var currentLatitude = 0.0
+    private var currentLongitude = 0.0
+    private lateinit var lastLocation: Location
+    private var mapMarker: Marker? = null
+    private lateinit var fusedLocationProviderClient: FusedLocationProviderClient
+    private lateinit var locationRequest: LocationRequest
+    private lateinit var locationCallback: LocationCallback
+    companion object {
+        private const val MY_PERMISSION_CODE: Int = 1000
+    }
+
+    //For Nearby Places
+    lateinit var googleAPIService: IGoogleAPIService
+    internal lateinit var nearbyPlaces: RootObject
+
+
+    override fun onMapReady(googleMap: GoogleMap) {
+        mMap = googleMap
+
+        //Init Google Play Services
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            if(ContextCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+                mMap!!.isMyLocationEnabled = true
+            }
+        } else {
+            mMap!!.isMyLocationEnabled = true
+        }
+
+        //Enable Zoom Control
+        mMap.uiSettings.isZoomControlsEnabled = true
+
+
+
+    }
+
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_map)
+
+        // Notify when the map is ready to be used.
+        val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as SupportMapFragment
+        mapFragment.getMapAsync(this)
+
+        //Init Service
+        googleAPIService = initiateGoogleAPIService.googleAPIService
+
+        //Request runtime permission
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            if (checkLocationPermission()) {
+                buildLocationRequest()
+                buildLocationCallBack()
+                fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this)
+                fusedLocationProviderClient.requestLocationUpdates(locationRequest, locationCallback, Looper.myLooper())
+            }
+        } else {
+            buildLocationRequest()
+            buildLocationCallBack()
+            fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this)
+            fusedLocationProviderClient.requestLocationUpdates(locationRequest, locationCallback, Looper.myLooper())
+        }
```

Comment:
This is a testament as to why we want a ViewModel. Every time this activity is created, you are re-running all these calls when they should be cached in some way.